### PR TITLE
Library Panels: Add library panel tab to share modal

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Adds the ability to add a panel to the panel library through a new tab on the panel share modal.

**Which issue(s) this PR fixes**:
Relates #31305

### CODE REVIEW

File: public/app/features/dashboard/components/ShareModal/ShareGlobalPanel.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+interface Props {
+  onDismiss?: () => void;
+  panel?: PanelModel;
+  initialFolderId?: number;
+}
+
+export const ShareGlobalPanel = ({ panel, initialFolderId, onDismiss }: Props) => {
+  if (!panel) {
+    return <></>;
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/ShareModal/ShareGlobalPanel.tsx:13]: Maybe return null here?

File: public/app/features/dashboard/components/ShareModal/types.ts [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
   dashboard: DashboardModel;
   panel?: PanelModel;
   onDismiss?(): void;
 }
 
-export type ShareModalTab = React.ComponentType<ShareModalTabProps>;
+type ShareModalTabPropsWithInternalModel = ShareModalTabProps & { panel?: InternalPanelModel };
+export type ShareModalTab =
+  | React.ComponentType<ShareModalTabProps>
+  | React.ComponentType<ShareModalTabPropsWithInternalModel>;
```
Comment (hugohaggmark) [inline public/app/features/dashboard/components/ShareModal/types.ts:14]: Hmm I wish we didn't need to reference the `InternalPanelModel` here, but I can see that would become hard to fix.

**Review by hugohaggmark — APPROVED** [review]
Looks great and works as expected 🎉

File: public/app/features/dashboard/components/ShareModal/types.ts (hunk 2 above)
Comment (kaydelaney) [reply public/app/features/dashboard/components/ShareModal/types.ts:14]: It's a bit strange actually, looking at some of the other Share modal components (`ShareSnapshot`, `ShareEmbed`, and `ShareExport`) all use the internal `PanelModel` for their `panel` prop, while the `ShareModalTab` type seems to use the `@grafana/data` `PanelModel` interface. Not exactly sure why typescript didn't complain about that, but did for the component I've introduced with this PR, but I suspect it has something to do with the fact that the component I've introduced is a function component whereas the others are all class components.

Comment (hugohaggmark) [reply public/app/features/dashboard/components/ShareModal/types.ts:14]: That's fine let's get this merged.

**Pull request merged**